Community/Tdarr_Plugin_Soggys_NVENC_HEVC_CQV_Optimised_Bitrate.js

```diff
@@ -0,0 +1,209 @@
+      type: 'number',
+      defaultValue: 28,
+      inputUI: {
+        type: 'text',
+      },
+      tooltip: `Enter the CQ:V value, lower is higher quality.
+
+      \\nExample:\\n
+      28`,
+    },
+    {
+      name: 'bframe',
+      type: 'number',
+      defaultValue: 0,
+      inputUI: {
+        type: 'text',
+      },
+      tooltip: `Specify amount of b-frames to use, 0-5. Use 0 to disable. (GPU must support this)
+
+      \\nExample:\\n
+      3`,
+    },
+    {
+      name: 'ten_bit',
+      type: 'string',
+      defaultValue: 'true',
+      inputUI: {
+        type: 'text',
+      },
+      tooltip: `Set to true to convert to 10-bit. false to disable
+
+      \\nExample:\\n
+      true`,
+    },
+    {
+      name: 'ffmpeg_preset',
+      type: 'string',
+      defaultValue: 'medium',
+      inputUI: {
+        type: 'text',
+      },
+      tooltip: `OPTIONAL, DEFAULTS TO MEDIUM IF NOT SET
+      \\n Enter the ffmpeg preset you want, leave blank for default (medium)
+
+      \\nExample:\\n
+        slow
+
+      \\nExample:\\n
+        medium
+
+      \\nExample:\\n
+        fast
+
+      \\nExample:\\n
+        veryfast`,
+    },
+  ],
+});
+
+// eslint-disable-next-line no-unused-vars
+const plugin = (file, librarySettings, inputs, otherArguments) => {
+  const lib = require('../methods/lib')();
+  // eslint-disable-next-line no-unused-vars,no-param-reassign
+  inputs = lib.loadDefaultValues(inputs, details);
+  let subcli = '-c:s copy';
+  let maxmux = '';
+  let map = '-map 0';
+  // default values that will be returned
+  const response = {
+    processFile: false,
+    preset: '',
+    container: '.mkv',
+    handBrakeMode: false,
+    FFmpegMode: false,
+    reQueueAfter: true,
+    infoLog: '',
+  };
+
+  const streamHeight = file.ffProbeData.streams[0].height;
+  const streamWidth = file.ffProbeData.streams[0].width;
+  const streamFPS = file.mediaInfo.track[0].FrameRate;
+  const streamBR = file.bit_rate;
+
+  // check if the file is a video, if not the function will be stopped immediately
+  if (file.fileMedium !== 'video') {
+    response.processFile = false;
+    response.infoLog += '☒File is not a video! \n';
+    return response;
+  }
+  response.infoLog += '☑File is a video! \n';
+
+  const optimalVideoBitrate = Math.floor((streamHeight * streamWidth * streamFPS) * inputs.targetCodecCompression);
+  response.infoLog += `Optimal Bitrate = ${(optimalVideoBitrate / 1000000).toString()}\n`;
+  response.infoLog += `Stream Bitrate = ${(streamBR / 1000000).toString()}\n`;
+  if (file.ffProbeData.streams[0].codec_name === 'hevc'
+  && (Number.isNaN(optimalVideoBitrate) || Number.isNaN(streamBR) || streamBR < optimalVideoBitrate)) {
+    response.infoLog += '☑File is already in hevc and is below optimal bitrate!\n';
+    return response;
+  }
+  if (file.ffProbeData.streams[0].codec_name === 'hevc'
+  && (file.video_resolution === '480p' || file.video_resolution === '576p' || file.video_resolution === '720p')) {
```

Comment:
You could also do this video_resolution comparison like `['480p', '576p', '720p'].includes(file.video_resolution)`. But it's just a preference thing, this works just fine.